common/ledger/util/leveldbhelper/leveldb_helper.go

```diff
@@ -205,16 +205,25 @@ func (f *FileLock) Lock() error {
 		panic(fmt.Sprintf("Error creating dir if missing: %s", err))
 	}
 	dbOpts.ErrorIfMissing = !dirEmpty
-	f.db, err = leveldb.OpenFile(f.filePath, dbOpts)
+	db, err := leveldb.OpenFile(f.filePath, dbOpts)
 	if err != nil && err == syscall.EAGAIN {
 		return errors.Errorf("lock is already acquired on file %s", f.filePath)
 	}
 	if err != nil {
 		panic(fmt.Sprintf("Error acquiring lock on file %s: %s", f.filePath, err))
 	}
+
+	// only mutate the lock db reference AFTER validating that the lock was held.
+	f.db = db
+
 	return nil
 }
 
+// Determine if the lock is currently held open.
+func (f *FileLock) IsLocked() bool {
+	return f.db != nil
+}
+
```

Comment:
I won't over-emphasize this but just to take a note - as the introduction of this function now assumes (at this level of code) that this struct can be used by a more than one go routines, access to field `f.db` should be protected.  Else, a caller can get a wrong result from this function. 

However, we are using this struct in a very limited way and don't foresee any plan to use this function in a test with concurrent go-routines, I am fine leaving it as is; but appreciate if you want to fix this for cleanliness at this level of the code.